jsm.go

```diff
@@ -611,6 +633,32 @@ type StreamLister struct {
 	pageInfo *apiPaged
 }
 
+// StreamsInfo returns a receive only channel to iterate on the streams.
+func (js *js) StreamsInfo(ctx context.Context) <-chan *StreamInfo {
+	if ctx == nil {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(context.Background(), js.wait)
+		defer cancel()
```

Comment:
I think this `defer` should be moved to be within the goroutine below, otherwise would get canceled right away.

Reply:
Whoops! Yeah.